Connectors/Rhino/Speckle.Connectors.RhinoShared/Bindings/RhinoMapperBinding.cs

```diff
@@ -17,60 +18,112 @@ public record CategoryMapping(
   int ObjectCount
 );
 
+/// <summary>
+/// Represents layers that are all assigned to the same category.
+/// </summary>
+public record LayerCategoryMapping(
+  string CategoryValue,
+  string CategoryLabel,
+  IReadOnlyList<string> LayerIds,
+  IReadOnlyList<string> LayerNames,
+  int LayerCount
+);
+
+/// <summary>
+/// Represents a layer option for the UI dropdown.
+/// </summary>
+public record LayerOption(string Id, string Name);
+
 /// <summary>
 /// Binding for managing Rhino object mappings to Revit categories.
 /// </summary>
 public class RhinoMapperBinding : IBinding
 {
+  private const string MAPPINGS_CHANGED_EVENT = "mappingsChanged";
+  private const string LAYERS_CHANGED_EVENT = "layersChanged";
   private readonly DocumentModelStore _store;
   private readonly IAppIdleManager _idleManager;
   private readonly IBasicConnectorBinding _basicConnectorBinding;
-  private const string CATEGORY_USER_STRING_KEY = "builtInCategory";
-  private const string MAPPINGS_CHANGED_EVENT = "mappingsChanged";
+  private readonly ITopLevelExceptionHandler _topLevelExceptionHandler;
+  private readonly RevitMappingResolver _revitMappingResolver;
   public string Name => "revitMapperBinding";
   public IBrowserBridge Parent { get; }
 
   public RhinoMapperBinding(
     DocumentModelStore store,
     IAppIdleManager idleManager,
     IBrowserBridge parent,
-    IBasicConnectorBinding basicConnectorBinding
+    IBasicConnectorBinding basicConnectorBinding,
+    ITopLevelExceptionHandler topLevelExceptionHandler,
+    RevitMappingResolver revitMappingResolver
   )
   {
     _store = store;
     _idleManager = idleManager;
     Parent = parent;
     _basicConnectorBinding = basicConnectorBinding;
+    _topLevelExceptionHandler = topLevelExceptionHandler;
+    _revitMappingResolver = revitMappingResolver;
 
     // Subscribe to Rhino events so we know about changes
     // Events fire on delete, undo delete and modify objects
     RhinoDoc.DeleteRhinoObject += OnObjectChanged;
     RhinoDoc.UndeleteRhinoObject += OnObjectChanged;
     RhinoDoc.ModifyObjectAttributes += OnObjectAttributesChanged;
 
+    // Subscribe to layer events so we know about layer changes
+    RhinoDoc.LayerTableEvent += OnLayerTableEvent;
+
     // Subscribe to document changes to refresh mappings when switching documents
     _store.DocumentChanged += OnDocumentChanged;
   }
 
-  #region UI Methods
+  #region UI Methods - General
 
   /// <summary>
   /// Gets list of available Revit categories for the UI dropdown.
   /// </summary>
   public CategoryOption[] GetAvailableCategories() =>
     RevitBuiltInCategoryStore.Categories.OrderBy(category => category.Label).ToArray();
 
+  /// <summary>
+  /// Gets list of available layers for the UI dropdown.
+  /// </summary>
+  public LayerOption[] GetAvailableLayers()
```

Comment:
can live on helper class, bindings tend to get large quickly, i would decrease noise here

Reply:
[refactor: consolidate Rhino layer and object utilities into helpers](https://github.com/specklesystems/speckle-sharp-connectors/pull/1027/commits/ee82d2435b27f47e210fc7aae8669e45b369c802)

Reply:
hmm, this was UI method, OK